provider/core.rb

```diff
@@ -239,31 +130,40 @@ def copy_file_list(output_folder, files)
       end.map(&:join)
     end
 
-    def compile_files(output_folder, version_name)
-      compile_file_list(output_folder, @config.files.compile, version_name)
+    # Compiles files specified within the product
+    def compile_product_files(output_folder, version_name)
+      file_template = ProductFileTemplate.new(
+        output_folder,
+        nil,
+        @api,
+        version_name,
+        build_env
+      )
+      compile_file_list(output_folder, @config.files.compile, file_template)
     end
 
-    def compile_common_files(output_folder, version_name = nil)
+    # Compiles files that are shared at the provider level
+    def compile_common_files(output_folder, version_name = nil, products = [])
```

Comment:
Since no callers use the optional arguments, there's no good reason to supply defaults imo (especially when a default for products will break Terraform)